client/src/components/addEventComponent.js

```diff
@@ -0,0 +1,250 @@
+import { logout, getCenters, apiLink } from '../reusables';
+import { connect } from 'react-redux';
+import { setEventDetail } from '../actions/eventActions';
+import { populateCenters } from '../actions/centerActions';
+
+const inputAttrs = (inputType, inputName, placeholder, className, ref, required) => ({
+  inputType,
+  inputName,
+  placeholder,
+  className,
+  ref,
+  required,
+});
+
+const mapDispatchToProps = dispatch => ({
+  setEventDetail: event => dispatch(setEventDetail(event)),
+  populateCenters: centers => dispatch(populateCenters(centers)),
+});
+const mapStateToProps = state => ({
+  modalTitle: state.page.modalTitle,
+  eventDetails: state.events.event,
+  eventDefaults: state.page.eventDefaults,
+  centers: state.centers.centerList,
+});
+let eventId;
+class AddEventComponent extends Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.folder = apiLink === 'http://localhost:8080' ? 'dev/events' : 'prod/events';
+  }
+  componentWillMount() {
+    getCenters(axios, this.props.populateCenters);
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    let postEvent;
+    const title = this.name.value;
+    const date = this.date.value;
+    const time = this.time.value;
+    const description = this.description.value;
+    const centerId = this.center.value;
+    const token = JSON.parse(localStorage.token).value;
+    const modalTitle = this.props.modalTitle;
+    const history = this.props.history;
+    const setEvent = this.props.setEventDetail;
+    function saveEvent(res) {
+      const eventData = {
+        title,
+        picture: res ? res.data.secure_url : undefined,
+        publicId: res ? res.data.public_id : undefined,
+        date,
+        time,
+        description,
+        centerId,
+        token,
+      };
+      if (modalTitle === 'New Event') {
+        postEvent = axios.post(`${apiLink}/api/v1/events`, eventData);
+      } else {
+        postEvent = axios.put(`${apiLink}/api/v1/events/${eventId}`, eventData);
+      }
+      postEvent
+        .then(response => {
+          history.push(`/dashboard/events/${response.data.data.id}`);
+          setEvent(response.data.data);
+          if (modalTitle !== 'New Event') {
+            $('#addNewEvent').modal('hide');
+          }
+        })
+        .catch(err => {
+          typeof err.response.data.message !== 'object' &&
+            alert(JSON.stringify(err.response.data.message));
+          (err.response.status === 403 || err.response.status === 401) &&
+            logout('addNewEvent', history);
+          if (typeof err.response.data.message === 'object') {
+            let occupiedDates = '';
+            err.response.data.message.OccupiedDates.forEach(date => {
+              occupiedDates += `${new Date(date).toDateString()}\n`;
+            });
+            alert(
+              `MESSAGE:\n${err.response.data.message.Sorry}\n\nOCCUPIED DATES:\n${occupiedDates}`
```

Comment:
String literals do not need `/n` for a new line. Just write your next line on the next line. It makes it more readable too.